packages/app/screens/landing/SignUpScreen.js

```diff
@@ -0,0 +1,335 @@
+            }}
+          />
+
+          <TextInput
+            testID="firstNameInput"
+            value={inputs["firstName"]}
+            container={{ marginBottom: 10 }}
+            style={[styles.component, styles.textInput]}
+            borderColor="#cccccc"
+            onChangeText={(text) => {
+              handleOnChange("firstName", text);
+              handleError("firstName", null);
+            }}
+            error={errors.firstName}
+            placeholder="First Name"
+            onSubmitEditing={validateInput}
+          />
+
+          <TextInput
+            testID="lastNameInput"
+            value={inputs["lastName"]}
+            container={{ marginBottom: 10 }}
+            style={[styles.component, styles.textInput]}
+            borderColor="#cccccc"
+            onChangeText={(text) => {
+              handleOnChange("lastName", text);
+              handleError("lastName", null);
+            }}
+            error={errors.lastName}
+            placeholder="Last Name"
+            onSubmitEditing={validateInput}
+          />
+
+          <TextInput
+            testID="passwordInput"
+            value={inputs["password"]}
+            container={{ marginBottom: 10 }}
+            ref={refPasswordInput}
+            style={[styles.component, styles.textInput]}
+            borderColor="#cccccc"
+            onChangeText={(text) => {
+              handleOnChange("password", text);
+              handleError("password", null);
+            }}
+            error={errors.password}
+            password
+            placeholder="Password"
+            onSubmitEditing={validateInput}
+          />
+
+          <TextInput
+            testID="passwordConfirmationInput"
+            value={inputs["passwordConfirmation"]}
+            container={{ marginBottom: 20 }}
+            ref={refPasswordConfirmationInput}
+            style={[styles.component, styles.textInput]}
+            borderColor="#cccccc"
+            onChangeText={(text) => {
+              handleOnChange("passwordConfirmation", text);
+              handleError("passwordConfirmation", null);
+            }}
+            error={errors.passwordConfirmation}
+            password
+            placeholder="Confirm Password"
+            onSubmitEditing={validateInput}
+          />
+
+          <Button
+            testID="signupButton"
+            title="Sign Up"
+            onPress={() => {
+              validateInput();
+            }}
+            underlayColor="#0e81c4"
+            fontColor="#ffffff"
+            fontWeight="bold"
+            style={[styles.loginButton, styles.component]}
+            textStyle={styles.boldText}
+          />
+
+          <DividerWithText title="or" />
+
+          <Button
+            testID="googleButton"
+            title="Continue with Google"
+            underlayColor="#ebebeb"
+            onPress={() => {}} //TODO: Implement Google Sign Up
```

Comment:
Signing up with Google is the same as signing in with Google for our case. We should extract the `handleOnLoginWithGoogle` function and export it to use in both the login and signup screen.